File: 백준/Silver/25757. 임스와 함께하는 미니게임/임스와 함께하는 미니게임.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        
        String[] input = br.readLine().trim().split(" ");
        int N = Integer.parseInt(input[0]);
        int div = 1;
        if(input[1].equals("F")) div = 2;
        else if(input[1].equals("O")) div = 3;
        Set<String> people = new HashSet<>();
        for(int i = 0; i < N; i++){
            people.add(br.readLine().trim());
        }
        System.out.println(people.size() / div);
    }
}

File: 백준/Silver/25757. 임스와 함께하는 미니게임/임스와 함께하는 미니게임.py
import sys

input = sys.stdin.readline

N, game = map(str, input().strip().split())

people = set()

for _ in range(int(N)):
    person = input().strip()
    people.add(person)

div = 1
if game == 'F':
    div = 2
elif game == 'O':
    div = 3

print(len(people) // div)